### ISSUE TITLE
Deprecator bug with setuptools >=71.0.3

### ISSUE THREAD

**Issue by AlexVCaron** [fury-gl/fury#912]
## Description

Installing `fury 0.10.0` with the new version of `setuptools (71.0.3)` succeeds. However, using the library fails if there is a call to a deprecated component. It comes from the usage of `setuptools.extern` to try to access `packaging`, which is not standard and discouraged by setuptools, see this issue pypa/setuptools#4498 . The problem seems to stem deeper, but note that on our side, everything was working correctly before `setuptools` version `71.0.3`.

## Stacktrace

```
Traceback (most recent call last):
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/pytest_console_scripts.py", line 187, in console_script
    s = entry_points[0].load()
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/pkg_resources/__init__.py", line 2749, in load
    return self.resolve()
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/pkg_resources/__init__.py", line 2755, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/runner/_work/scilpy/scilpy/scripts/scil_bundle_diameter.py", line 35, in <module>
    from fury import actor
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/actor.py", line 12, in <module>
    from fury.actors.peak import PeakActor
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/actors/peak.py", line 16, in <module>
    from fury.shaders import (
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/shaders/__init__.py", line 1, in <module>
    from fury.shaders.base import (
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/shaders/base.py", line 140, in <module>
    @deprecate_with_version(
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/deprecator.py", line 189, in deprecate_with_version
    'Raises' if is_bad_version(until) else 'Will raise', error_class, until
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/deprecator.py", line 132, in is_bad_version
    return version_comparator(version_str) == -1
  File "/home/runner/_work/_tool/Python/3.10.14/x64/lib/python3.10/site-packages/fury/deprecator.py", line 122, in cmp_pkg_version
    elif version_cmp(version_str) > version_cmp(pkg_version_str):
TypeError: 'NoneType' object is not callable
```

## Way to reproduce

1. `pip install setuptools==71.0.3 fury==0.10.0`
2. Create a small python script using this import : `from fury import actor`
3. Run the script, it should reproduce the error

## OS and versions
```
{'fury_version': '0.10.0', 'pkg_path': '/home/scil/dev/scilpy/venv/lib/python3.10/site-packages/fury', 'commit_hash': '83aab2d7', 'sys_version': '3.10.12 (main, Mar 22 2024, 16:50:05) [GCC 11.4.0]', 'sys_executable': '/home/scil/dev/scilpy/venv/bin/python', 'sys_platform': 'linux', 'numpy_version': '1.23.5', 'scipy_version': '1.9.3', 'vtk_version': '9.2.6', 'matplotlib_version': '3.6.3', 'dipy_version': '1.9.0'}
```

**Comment by skoudoro** [comment]
Thank you for this report.

Personally, I'm in a family break so I will be able to look into it next week.

Maybe @WassCodeur can look into it.

**Comment by WassCodeur** [comment]
Hey @skoudoro  @AlexVCaron 

I was able to reproduce the error,

now I see the source of the error and I'm trying to fix it and open a pr.

**Comment by WassCodeur** [comment]
Hi @skoudoro !

PLease can I close this issue ?

**Comment by skoudoro** [comment]
closed by #913 

it should be on the release of this week @AlexVCaron.
